MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/InstrumentView/UnwrappedSurface.h

```diff
@@ -44,17 +44,23 @@ class UnwrappedDetector {
 public:
   UnwrappedDetector();
   UnwrappedDetector(const unsigned char *c,
-                    boost::shared_ptr<const Mantid::Geometry::IDetector> det);
+                    const Mantid::Geometry::IDetector &det);
   UnwrappedDetector(const UnwrappedDetector &other);
   UnwrappedDetector &operator=(const UnwrappedDetector &other);
+  bool isValid() const;
   unsigned char color[3]; ///< red, green, blue colour components (0 - 255)
   double u;               ///< horizontal "unwrapped" coordinate
   double v;               ///< vertical "unwrapped" coordinate
   double width;           ///< detector width in units of u
   double height;          ///< detector height in units of v
   double uscale;          ///< scaling factor in u direction
   double vscale;          ///< scaling factor in v direction
-  Mantid::Geometry::IDetector_const_sptr detector;
+  Mantid::detid_t detID;  ///< Detector ID
+  Mantid::Kernel::V3D position;  ///< Detector position
+  Mantid::Kernel::Quat rotation; ///< Detector orientation
+  boost::shared_ptr<const Mantid::Geometry::Object>
+      shape;                       ///< Shape of the detector
+  Mantid::Kernel::V3D scaleFactor; ///< Detector's scale factor
```

Comment:
The values cached here may actually be different from what is found in the instrument? Or is this just a caching mechanism for speed?

`DetectorInfo` would eventually be a fast way for accessing these paramters, but currently it is not yet, so I think I agree that it makes sense having this.